projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -449,11 +445,38 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
     private onValidatorChange: () => void = noop;
 
     constructor(public element: ElementRef,
-        @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions: IDisplayDensityOptions,
-        @Inject(LOCALE_ID) private localeId: any,
-        private _injector: Injector) {
-        super(_displayDensityOptions);
-        this.locale = this.locale || this.localeId;
+        @Inject(LOCALE_ID) protected _localeId: any,
+        private _injector: Injector,
+        private _moduleRef: NgModuleRef<any>,
+        @Inject(IgxOverlayService) private _overlayService: IgxOverlayService,
+        @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions?: IDisplayDensityOptions,
+        @Optional() @Inject(IGX_INPUT_GROUP_TYPE) protected _inputGroupType?: IgxInputGroupType) {
+        super(element, _localeId, _displayDensityOptions, _inputGroupType);
+        this.locale = this.locale || this._localeId;
+    }
+
+    /** @hidden @internal */
+    @HostListener('keydown', ['$event'])
+    /** @hidden @internal */
+    public onKeyDown(event: KeyboardEvent): void {
+        switch (event.key) {
+            case KEYS.UP_ARROW:
+            case KEYS.UP_ARROW_IE:
+                if (event.altKey) {
+                    this.close();
+                }
+                break;
+            case KEYS.DOWN_ARROW:
+            case KEYS.DOWN_ARROW_IE:
+                if (event.altKey) {
+                    this.open();
+                }
+                break;
+            case KEYS.ESCAPE:
```

Comment:
you can set `closeOnEscape: true` for both dialog and drop down overlay settings, and do not handle `ESC`